mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMailboxMapper.java

```diff
@@ -134,16 +136,10 @@ public void endRequest() {
     /**
      * @see org.apache.james.mailbox.store.mail.MailboxMapper#hasChildren(org.apache.james.mailbox.store.mail.model.Mailbox, char)
      */
-    public boolean hasChildren(Mailbox mailbox, char delimiter) throws MailboxException {
-        String mailboxName = mailbox.getName() + delimiter;
+    public boolean hasChildren(Mailbox mailbox, PathDelimiter delimiter) throws MailboxException {
         return mailboxesByPath.values()
             .stream()
-            .anyMatch(box -> belongsToSameUser(mailbox, box) && box.getName().startsWith(mailboxName));
```

Comment:
you don't check the user anymore ?

Reply:
The user check is handled by is child of.